### PR TITLE
Support anonymous authentication

### PR DESCRIPTION
**Pull request by zdtsw**
In the current implementation, it requires a config file (by default auth.props) to have user and pass set. Otherwise, application return 401. Thinking it might not really need to have a user, if we can directly query from website as non-logined user, it should not be difference to query these information as anonymous with Jrjc client. 
This PR is intend to make both ways work: 
- when auth.props file has user/pass, use BasicHttpAuthenticationHandler
- when auth.props is empty, use AnonymousAuthenticationHandler
- when no auth.props file exists, use AnonymousAuthenticationHandler

### CODE REVIEW

**Comment by shipilev** [comment]
> In the current implementation, it requires a config file (by default auth.props) to have user and pass set. Otherwise, application return 401. Thinking it might not really need to have a user, if we can directly query from website as non-logined user, it should not be difference to query these information as anonymous with Jrjc client. This PR is intend to make both ways work:

The current auth mechanism was done after observing that OpenJDK JIRA does not allow some queries for anonymous user. For example, try this: `java -jar jdk-backports-monitor.java --affiliation` with and without auth.

But, I agree we might want to support anonymous auth as well.

File: README.md [hunk 1]
```diff
@@ -21,7 +21,9 @@ seen the Bash version I had before it.
 
 Many queries involve complex JQL statements, which requires authentication to execute. 
 The tool expects the property file with your user/login for OpenJDK bug database
-(comes with a glorious OpenJDK Author role, after some contributions):
+(comes with a glorious OpenJDK Author role, after some contributions).
+By default, application uses a file "auth.props" with user/pass as authenticated method.
+An empty auth.props file will switch to use anonymous authentication.
```
Comment (shipilev) [inline README.md]: I think we also need to support the case of non-existent `auth.props`, and fall back to anonymous authentication as well. This relaxes the "tool expects the property file..." sentence above as well.

File: src/main/java/org/openjdk/backports/jira/Connect.java [hunk 2]
```diff
@@ -77,10 +79,12 @@ public static Clients getClients(String jiraURL, String user, String pass) throw
...
         HttpClient client = factory.create(opts);
 
-        DisposableHttpClient dispClient = new AtlassianHttpClientDecorator(
-                client,
-                new BasicHttpAuthenticationHandler(user, pass)) {
-                    @Override public void destroy() throws Exception { factory.dispose(client); }
+        AuthenticationHandler auth = new AnonymousAuthenticationHandler();
+        if (user != null && pass != null) {
+            auth = new BasicHttpAuthenticationHandler(user, pass);
+        }
```
Comment (shipilev) [inline src/main/java/org/openjdk/backports/jira/Connect.java]:
```suggestion
        AuthenticationHandler auth;
        if (user != null && pass != null) {
            auth = new BasicHttpAuthenticationHandler(user, pass);
        } else {
            auth = new AnonymousAuthenticationHandler();
        }
```

**Review by shipilev — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (zdtsw) [reply README.md]: Yes, I agree, just thought that would be a different feature to handle `--auth` file, then should be in a different PR.

Comment (shipilev) [reply README.md]: No need, we can cobble all related changes in this PR.

**Comment by zdtsw** [comment]
> > In the current implementation, it requires a config file (by default auth.props) to have user and pass set. Otherwise, application return 401. Thinking it might not really need to have a user, if we can directly query from website as non-logined user, it should not be difference to query these information as anonymous with Jrjc client. This PR is intend to make both ways work:
> 
> The current auth mechanism was done after observing that OpenJDK JIRA does not allow some queries for anonymous user. For example, try this: `java -jar jdk-backports-monitor.java --affiliation` with and without auth.
> 
> But, I agree we might want to support anonymous auth as well.

Yeah, the result is different to show `Unknown` for all users if run as "anonymous"

The main reason to add this feature is because Adoptium might want to use this application. but impossible to do this  without a service account. the use-case is quite simple to only query Jira with version/label etc.

**Review by shipilev — APPROVED** [review]
Looks good!

**Pull request merged**